### PR TITLE
chore: Fix unused variables

### PR DESCRIPTION
**Pull request by efritz**
Appeases new linting rules for Go.

## Test plan

N/A.

### CODE REVIEW

File: internal/codeintel/policies/service.go [hunk 1]
```diff
@@ -40,6 +40,7 @@ func (s *Service) Get(ctx context.Context, id int) (policy Policy, ok bool, err
 	defer endObservation(1, observation.Args{})
 
 	// To be implemented in https://github.com/sourcegraph/sourcegraph/issues/33376
+	_ = ctx
```
Comment (bobheadxi) [inline internal/codeintel/policies/service.go:43]: Isn't the better fix here:

```go
_, endObservation := s.operations.get.With(ctx, &err, observation.Args{})
```

**Review by bobheadxi — APPROVED** [review]
dont want to block a fix tho

**Comment by bobheadxi** [comment]
> new linting rules for Go.

* _re-instated_ 😛

File: internal/codeintel/policies/service.go (hunk 1 above)
Comment (efritz) [reply internal/codeintel/policies/service.go:43]: #33376 & co are next on the table to be implemented so whatever fix we do here won't last long

**Pull request merged**